### ISSUE TITLE
Generated class is not importing an enum

### ISSUE THREAD

**Issue by zhedi** [apollographql/apollo-android#3264]
I know there is another issue #1076 on the same topic, but I've tried the workaround by moving the graphql files into a different package but it didn't work. I've tried with both client version v1.x and v2.x, neither worked.

Error msg is:
```bash
 [javac] /SomeRootPath/build/private/apollogql/src/com/something/mysomethingservice/servername/GetXSummaryQuery.java:1695: error: cannot find symbol
    [javac]                 listItemWriter.writeString(((XStage) item).rawValue());
    [javac]                                                               ^
    [javac]   symbol:   method rawValue()
    [javac]   location: class XStage
    [javac] /SomeRootPath/build/private/apollogql/src/com/something/mysomethingservice/servername/GetXSummaryQuery.java:1783: error: cannot find symbol
    [javac]             return XStage.safeValueOf(listItemReader.readString());
    [javac]                              ^
    [javac]   symbol:   method safeValueOf(String)
    [javac]   location: class XStage
    [javac] 2 errors
```

The relevant graphql defs are:
```graphql
type DailyXSummary {
    xArchitecture: XArchitectureInfo
    xQuality: XQuality!
}

type XArchitectureInfo {
  someNotImportantField: String!
  xStages: [xProperty!]!
}

type xProperty {
  xStage: XStage!
}

enum XStage {
  Unknown
  HappyStage
  SadStage
  OkStage
}

enum XQuality {
    Good
    Poor
}

query GetXSummaryQuery(
    $includeDailyXSummary: Boolean!
    $timeStringDay: String!
) {
    dailyXSummary: xSummary(input: { timeString: $timeStringDay, period: Day, presentationOptions: { exactValuesOverride: false }})
        @include(if: $includeDailyXSummary) {
            id
            ... on DailyXSummary {
                xArchitecture {
                    someNotImportantField
                    xStages {
                        xStage
                    }
                }
                xQuality
            }
        }
}
```

`XStage.java` was generated properly in the path: build/generated-src/com/something/somethingservice/servername/type with content
```java
// AUTO-GENERATED FILE. DO NOT MODIFY.
//
// This class was automatically generated by Apollo GraphQL plugin from the GraphQL queries it found.
// It should not be modified by hand.
//
package com.something.somethingservice.servername.type;

import java.lang.String;

public enum XStage {
  HappyStage("Happy"),

  SadStage("Sad"),

  OkStage("Ok"),

  /**
   * Auto generated constant for unknown enum values
   */
  $UNKNOWN("$UNKNOWN");

  private final String rawValue;

  XStage(String rawValue) {
    this.rawValue = rawValue;
  }

  public String rawValue() {
    return rawValue;
  }

  public static XStage safeValueOf(String rawValue) {
    for (XStage enumValue : values()) {
      if (enumValue.rawValue.equals(rawValue)) {
        return enumValue;
      }
    }
    return XStage.$UNKNOWN;
  }
}
```

However, `XStage.java` was not actually imported into the generated `GetXSummaryQuery.java` file (which has the path: build/generated-src/com/something/somethingservice/servername/)

Due to confidentiality reasons, I can't actually paste the content of `GetXSummaryQuery.java` (it is too hard to manually redact that to be in sync with what I have above since the original schemas are much larger and therefore the generated `GetXSummaryQuery.java` is very very long)

But the relevant import statements are as follows:
```java
package com.something.somethingservice.servername;

import com.something.somethingservice.servername.type.XQuality;
import com.apollographql.apollo.api.FragmentResponseFieldMapper;
import com.apollographql.apollo.api.InputFieldMarshaller;
import com.apollographql.apollo.api.InputFieldWriter;
```

Basically, `XQuality` as an Enum is properly imported whereas `XStage` was not. `XStage` does appear in the `GetXSummaryQuery.java` code but was not imported. 

My guess is that the issue has something to do with the nesting structure `XStage` is in. (Reason: both `XStage` and `XQuality` are Enums with string values; `XStage` is in a nesting structure in the query but `XQuality` isn't)

The query itself works properly and returns correct data while testing in graphql clients such as GraphQL Playground.

**Comment by martinbonnin** [comment]
Thanks for reaching out and for the detailed information! I put together a reproducer there: https://github.com/martinbonnin/reproducer-3264/blob/main/src/main/graphql/operations.graphql#L12

You guessed right, this happends because the nested `XStage` conflicts with the `XStage` enum. 
You can workaround this by using aliases:

```graphql
      xArchitecture {
        someNotImportantField
        # xStages conflicts with the enum below. Use an alias to workaround this
        dailyXStages: xStages {
          xStage
        }
      }
```

**Comment by zhedi** [comment]
Thanks a lot Martin. The workaround worked perfectly!

**Comment by surabhi-raje** [comment]
Thanks a lot Martin for the workaround! This worked :) I'm not sure why "xStages" conflicts with "xStage", but it would be great if we could add a note regarding this in the documentation for apollo-android; or if possible, fix the compilation logic to not have the name conflict.

**Comment by martinbonnin** [comment]
@surabhi-raje agreed. I don't think we can automatically remove these nameclashes but we should be able to provide better errors when that happens. I opened https://github.com/apollographql/apollo-android/issues/3283 as a follow up
